core/src/client/components/DatabaseConfigurationModal.tsx

```diff
@@ -87,6 +101,8 @@ export default class DatabaseConfigurationModal extends PureComponent<Props, Sta
                 </Modal.Header>
 
                 <Modal.Body>
+                    {hasError && <Alert bsStyle="danger">{error || initError}</Alert>}
```

Comment:
would there ever be a case where both errors would exist? If so, I could see having something like this be sufficient to catch the either-or case:
```suggestion
                    {error && <Alert bsStyle="danger">{error}</Alert>}
                    {initError && <Alert bsStyle="danger">{initError}</Alert>}
```

Reply:
I don't think so. Init errors will only occur on `componentDidMount` and i prevented update in the case that there are init errors since we shouldn't allow mutation if we don't have current server props.